src/oca_github_bot/manifest.py

```diff
@@ -234,6 +246,43 @@ def user_can_push(gh, org, repo, username, addons_dir, target_branch):
     current_branch = git_get_current_branch(cwd=addons_dir)
     try:
         check_call(["git", "checkout", target_branch], cwd=addons_dir)
-        return is_maintainer(username, modified_addon_dirs)
+        result = is_maintainer(username, modified_addon_dirs)
     finally:
         check_call(["git", "checkout", current_branch], cwd=addons_dir)
+
+    if result:
+        return True
+
+    other_branches = config.MAINTAINER_CHECK_ODOO_RELEASES
+    if target_branch in other_branches:
+        other_branches.remove(target_branch)
+
+    return is_maintainer_other_branches(
+        org, repo, username, modified_addons, other_branches
+    )
+
+
+def is_maintainer_other_branches(org, repo, username, modified_addons, other_branches):
+    for addon in modified_addons:
+        is_maintainer = False
+        for branch in other_branches:
+            manifest_file = (
+                float(branch) < 10.0 and "__openerp__.py" or "__manifest__.py"
+            )
+            url = (
+                f"https://raw.githubusercontent.com/{org}/{repo}/{branch}/"
+                f"{addon}/{manifest_file}"
+            )
+            _logger.debug("Looking for maintainers in %s" % url)
+            r = requests.get(
+                url, allow_redirects=True, headers={"Cache-Control": "no-cache"}
+            )
+            if r.ok:
+                manifest = ast.literal_eval(r.content.decode("utf-8"))
```

Comment:
nit: it might be better to add a `parse_manifest(manifest: bytes)` function in `manifest.py`.